corehq/apps/hqwebapp/static/hqwebapp/js/inactivity.js

```diff
@@ -110,16 +120,38 @@ hqDefine('hqwebapp/js/inactivity', [
             shouldShowWarning = false;
         };
 
+        var isModalOpen = function (element) {
+            // https://stackoverflow.com/questions/19506672/how-to-check-if-bootstrap-modal-is-open-so-i-can-use-jquery-validate
+            return (element.data('bs.modal') || {}).isShown;
+        };
+
+        var showPageRefreshModal = function () {
+            if (!isModalOpen($modal) && !isModalOpen($warningModal)) {
+                $newVersionModal.modal('show');
+            }
+        };
+
         var pollToShowModal = function () {
             log("polling HQ's ping_login to decide about showing login modal");
+            var selectedAppId = '';
+            try {
+                var urlParams = JSON.parse(decodeURIComponent(window.location.hash.substr(1)));
+                selectedAppId = urlParams.appId;
+            } catch (error) {
+                console.log(error);
```

Comment:
Could/should this post to `/jserror/` so we get notified?

Reply:
I am not sure if we would be getting anything out of this error. For the pages which do not have appId in URL(like the apps page), we expect to see an error in `JSON.parse`. Do you think we should be sending them out to Sentry?

Reply:
Oh, I get it, I wasn't considering that this is going to fail often (though less often if you move the modal to web apps and check if it exists). Anyway, nah, no need to send to sentry.